src/test.rs

```diff
@@ -501,3 +501,69 @@ fn var_drop_used_simple() {
         tables.points.untern(first_defined)
     );
 }
+
+/// This test ensures one of the two placeholder origins will flow into the
+/// other, but without declaring this subset as a `known_subset`, which is
+/// an illegal subset relation error.
+#[test]
+fn illegal_subset_error() {
+    let program = r"
+        universal_regions { 'a, 'b }
+        placeholder_loans { 'a: L1, 'b: L2 }
+        
+        block B0 {
+            // creates a transitive `'b: 'a` subset
+            borrow_region_at('x, L3),
+              outlives('b: 'x),
+              outlives('x: 'a);
+        }
+    ";
+
+    let mut tables = intern::InternerTables::new();
+    let facts = parse_from_program(program, &mut tables).expect("Parsing failure");
+
+    assert_eq!(facts.universal_region.len(), 2);
+    assert_eq!(facts.placeholder_loan.len(), 2);
+
+    // no known subsets are defined in the program...
+    assert_eq!(facts.known_subset.len(), 0);
+
+    let result = Output::compute(&facts, Algorithm::Naive, true);
+
+    // ...so there should be an error here about the missing `'b: 'a` subset
+    assert_eq!(result.subset_errors.len(), 1);
+
+    let point = Point::from(1);
+    let subset_error = result.subset_errors.get(&point).unwrap();
+
+    let origin_a = Origin::from(0);
```

Comment:
as an aside, a nice cleanup would be if we could produce "user-readable" errors and diff the output from those in these tests, no?

Reply:
yes I'd like this a lot, including reorganizing tests

Reply:
I've made a couple attempts at this, and they're now more readable thanks to a `FactChecker` assertion helper. In https://github.com/rust-lang/polonius/pull/137/commits/0db817d4de435a28c21c863589a13d7cef8bf46a#diff-c5fc4beef0f03493c2e9a9c876073f0fR535 there's an example where the assertions use the regular atom values and not the interned ones.